File: others/printer_jobs.py
def jobtime(jobs):
    epoch = 0
    rest = 2
    d = {}
    for job in jobs:
        if job in d.keys():
            delta_t = epoch - d[job]
            cool = rest + 1 - delta_t
            if cool > 0:
                epoch += cool

        d[job] = epoch
        epoch += 1

    return epoch

if __name__ == '__main__':
    assert jobtime('AAA') == 7
    assert jobtime('') == 0
    assert jobtime('ABA') == 4

File: others/printer_jobs.cpp
#include <string>
#include <iostream>
#include <unordered_map>

using namespace std;

int jobtime(string jobs) {
    unordered_map<char, int> d;
    int epoch = 0;
    int rest = 2;
    
    for (char ch : jobs) {
        if (d.find(ch) != d.end()) {

            int delta_t = epoch - d[ch];
            int cool = rest + 1 - delta_t;
            if (cool > 0) {
                epoch += cool;

            }
        }
        d[ch] = epoch;
        epoch += 1;
    }
    return epoch;
}

int main() {
    cout << "AAA = " << jobtime("AAA") << endl;
    cout << "ABA = " << jobtime("ABA") << endl;
    return 0;
}
